### PR TITLE
Hotfix branch support

### PR DESCRIPTION
**Pull request by y-asaba**
Add hotfix command for hotfix branches.
- Hotfix branches?
  Hotfix branches are for critical fixes. Please refer the following URL
  for more detail.
  http://nvie.com/posts/a-successful-git-branching-model/
- Behavior
  *\* git daily hotfix open
- Checks out the master branch
- Creates a new branch from the master branch.

*\* git daily hotfix close
1. Merges diffs on a hotfix branch into the master branch.
2.1. Merges diffs into the develop branch if a release branch is not open.
2.2. Merges diffs into a release branch if release is open.
3. Delete the hotfix branch

*\* other sub commands
Same as subcommands of the release command.
- Design
  The hotfix command is similar to the release command.
  So the new Git_Daily_Command_Hotfix class inherits
  Git_Daily_Command_Release class and changes some variables.
- Usage
  Usage: git daily hotfix open        : Open hotfix-release process
  or: git daily hotfix list        : Show hotfix list
  or: git daily hotfix sync        : Sync current opened hotfix process
  or: git daily hotfix close       : Close to hotfix-release process
- Test
  cd test/hotfix
  ./hotfix-test1
  ./hotfix-test2

I would appreciate it if you could review this pull request.
Thanks.

### CODE REVIEW

File: src/Git/Daily/Command/Hotfix.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    protected $option = array(
+    );
+
+    protected $base_branch = 'master';
+
+    protected $merge_to = array('master', 'develop', 'release');
+
+    protected $branch_prefix = 'hotfix';
+
+    function getMergeBranches()
```
Comment (sotarok) [inline src/Git/Daily/Command/Hotfix.php]: modifier ?

File: src/Git/Daily/Command/Release.php [hunk 2]
```diff
@@ -19,6 +19,20 @@ class Git_Daily_Command_Release
...
+    protected $base_branch = 'develop';
+
+    protected $merge_to = array('master', 'develop');
+
+    protected $branch_prefix = 'release';
+
+    /**
+     *
+     */
+    function getMergeBranches()
```
Comment (sotarok) [inline src/Git/Daily/Command/Release.php]: modifier ?

File: src/Git/Daily/Command/Release.php [hunk 3]
```diff
@@ -286,98 +300,75 @@ class Git_Daily_Command_Release
...
-            self::outLn($res);
-            if ($retval != 0) {
-                self::warn("$master_branch pull failed");
-                throw new Git_Daily_Exception('abort');
+        //
+        // Start merging diffs
+        //
+        foreach ($this->getMergeBranches() as $branch_name) {
+            $merge_branches = self::cmd(Git_Daily::$git, array('branch'),
+                                      array('grep', array($branch_name), array('sed', array('s/^[^a-zA-Z0-9]*//g'),
```
Comment (sotarok) [inline src/Git/Daily/Command/Release.php]: this indent style is not same as mine

**Comment by sotarok** [comment]
Thank you for your request.

This is great feature and I wished it but I cannot merge this if this goes on. So please check following points:
1. Please rebase my latest develop branch, you degraded my commit like 47eb13d7542b39bac8e36399f5e2906592ed1b14 .
2. Please fix some different coding styles with me. I commented here: https://github.com/sotarok/git-daily/pull/8/files
3. Please rebase several commits to one commit and re-push to this branch (you can push local rebased branch with following command: `git push -f origin feature/hotfix` )

**Comment by y-asaba** [comment]
Thank you for your feedback.
I have updated this pull request. Could you review this again?

Thanks.

**Pull request merged**